src/docfx/lib/schema/JsonSchemaValidation.cs

```diff
@@ -12,12 +12,14 @@ internal static class JsonSchemaValidation
         public static List<Error> Validate(JsonSchema schema, JToken token)
         {
             var errors = new List<Error>();
-            Validate(schema, token, errors);
+            Validate(schema, token, errors, schema);
             return errors;
         }
 
-        private static void Validate(JsonSchema schema, JToken token, List<Error> errors)
+        private static void Validate(JsonSchema schema, JToken token, List<Error> errors, JsonSchema root)
```

Comment:
consider having a `JsonSchemaValidationContext` class that holds the logic of JsonSchemaUtility, we may need instance fields.

```suggestion
        private static void Validate(JsonSchemaValidationContext context, JsonSchema schema, JToken token, List<Error> errors)
```